### PR TITLE
Wrapped cli_context with function

### PR DESCRIPTION
**Pull request by sfc-gh-astus**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Wrapped cli_context with function. 
Creating global variables in that way case initialize them during module loading, which can cause problems. We faced one this week  in our tests with config manager from Python connector.

### CODE REVIEW

**Review by sfc-gh-jvasquezrojas — APPROVED** [review]

File: src/snowflake/cli/api/cli_global_context.py [hunk 1]
```diff
@@ -386,5 +386,19 @@ def _should_force_mute_intermediate_output(self) -> bool:
         return self._manager.output_format == OutputFormat.JSON
 
 
-cli_context_manager: _CliGlobalContextManager = _CliGlobalContextManager()
-cli_context: _CliGlobalContextAccess = _CliGlobalContextAccess(cli_context_manager)
+_CLI_CONTEXT_MANAGER: _CliGlobalContextManager | None = None
+_CLI_CONTEXT: _CliGlobalContextAccess | None = None
+
+
+def get_cli_context_manager() -> _CliGlobalContextManager:
```
Comment (sfc-gh-melnacouzi) [inline src/snowflake/cli/api/cli_global_context.py:393]: I am curious, what is the reason for this change? (Would help to add it in the PR description as well)

Comment (sfc-gh-astus) [reply src/snowflake/cli/api/cli_global_context.py:393]: Added

**Review by sfc-gh-pczajka — DISMISSED** [review]

**Review by sfc-gh-turbaszek — DISMISSED** [review]

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Review by sfc-gh-turbaszek — APPROVED** [review]
